lib/internal/streams/pipeline.js

```diff
@@ -25,10 +25,10 @@ let EE;
 let PassThrough;
 let createReadableStreamAsyncIterator;
 
-function destroyer(stream, reading, writing, final, callback) {
+function destroyer(stream, reading, writing, callback) {
   const _destroy = once((err) => {
     const readable = stream.readable || isRequest(stream);
-    if (err || !final || !readable) {
```

Comment:
You still need final here. Final here means the last stream. Not necessarily the last one to finish.

Reply:
Giving this more thought, I'm a little unsure whether final is needed here or not. Might be fine without.

Reply:
Giving it even more thought, it's probably better to leave it as it was. Legacy streams might not be properly handling `stream.readable` so I would prefer to avoid applying that logic where possible.

Reply:
Done.